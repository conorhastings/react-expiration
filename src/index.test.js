import React from "react";
import renderer from "react-test-renderer";
import Expiration from "./index.js";

/* note: this test will fail in the year 3000 */
test("Expiration Component should render not expired component if current date is before expirationDate", () => {
  const tree = renderer
    .create(
      <Expiration
        render={isExpired => isExpired ? "expired" : "not expired"}
        expirationDate={new Date("12/31/2999")}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* note: this test might fail if there's ever time travel and you run these thests in the 90s */
test("Expiration Component should render expired component if current date is after expirationDate", () => {
  const tree = renderer
    .create(
      <Expiration
        render={isExpired => (isExpired ? "expired" : "not expired")}
        expirationDate={new Date("12/31/1999")}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
