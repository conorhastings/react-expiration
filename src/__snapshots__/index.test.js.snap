// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expiration Component should render expired component if current date is after expirationDate 1`] = `"expired"`;

exports[`Expiration Component should render not expired component if current date is before expirationDate 1`] = `"not expired"`;
